File: TripletSum/TripletSum.py
def TripletSum(arr , sum):
     l = 0
     r = 0
     arr.sort()
     for i in range(0, len(arr)-2):
         l = i + 1
         r = len(arr)-1
         while (l < r):
            if( arr[i] + arr[l] + arr[r] == sum):
                print("Triplet is", arr[i],
                     ', ', arr[l], ', ', arr[r])
                return True
            elif (arr[i] + arr[l] + arr[r] < sum):
                l += 1
            else:
                r -= 1
     return False
     
    


TripletSum([1, 4, 45, 6, 10, 8] , 22)

File: TripletSum/TripletSum.js
function TripletSum(arr, sum) {
  let l, r;
  arr.sort((a, b) => a - b);
  for (let i = 0; i < arr.length - 2; i++) {
    l = i + 1;
    r = arr.length - 1;
    while (l < r) {
      if (arr[i] + arr[l] + arr[r] == sum) {
        console.log("Triplet is " + arr[i] + ", " + arr[l] + ", " + arr[r]);
        return true;
      } else if (arr[i] + arr[l] + arr[r] < sum) l++;
      // A[i] + A[l] + A[r] > sum
      else r--;
    }
  }
  return false;
}

//Driver Code

TripletSum([1, 4, 45, 6, 10, 8], 22);
